src/faucet/views.rs

```diff
@@ -172,13 +166,26 @@ pub fn Faucet(target_network: Network) -> impl IntoView {
                         }.into_any()
                     }
                 }}
-
             </div>
             <div class="flex justify-between my-4">
                 <div>
                     <h3 class="text-lg font-semibold">Faucet Balance:</h3>
                     <Transition fallback={move || view!{ <p>Loading faucet balance...</p> }}>
-                        <p class="text-xl">{ move || format_balance(&faucet.get().get_faucet_balance(), &faucet.get().get_fil_unit()) }</p>
+                    {move || {
+                        if faucet.get().is_low_balance() {
+                            view! {
+                                <a class="text-blue-600" target="_blank" rel="noopener noreferrer" href={topup_req_url}>
+                                    "Request Faucet Top-up"
+                                </a>
+                            }.into_any()
+                        } else {
+                            view! {
+                                <p class="text-xl">
+                                    { format_balance(&faucet.get().get_faucet_balance(), &faucet.get().get_fil_unit()) }
+                                </p>
+                            }.into_any()
+                        }
+                    }}
```

Comment:
It'd be cleaner to detach this from the main logic and have a separate component.

Reply:
@LesnyRumcajs right the `Faucet` component itself has grown big, we can break it into smaller meaningful components.
better to refactor this in a separate PR

Reply:
I'm not asking for a full refactor here (though it might be needed, ultimately). I'm a big fan of small changes and the _boy scout rule_. Just extracting this small component is a step in the right direction.

> The "Scout Rule" in programming, also known as the "Boy Scout Rule", is a principle that encourages developers to leave code cleaner and more maintainable than they found it. It's inspired by the Boy Scouts' motto of leaving campsites cleaner than they were when they arrived. In essence, it promotes a culture of continuous improvement and responsibility within a codebase

Reply:
Done